File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int buyLow = INT_MAX, sellHigh = 0;
        
        for (int p : prices) {
            
            buyLow = min(buyLow, p);
            
            sellHigh = max(sellHigh, p - buyLow);
            
            //cout << "low: " << buyLow << ", high: " << sellHigh << endl;
        }
        
        return sellHigh;
    }
};

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int buyLow = Integer.MAX_VALUE, sellHigh = 0;
        
        for (int p : prices) {
            
            buyLow = Math.min(buyLow, p);
            
            sellHigh = Math.max(sellHigh, p - buyLow);
            
            //System.out.println("low: " + buyLow + ", high: " + sellHigh);
        }
        
        return sellHigh;
    }
}